File: sample-app/src/main/java/io/codegik/sample/InductionHeaders.java
package io.codegik.sample;

/**
 * Header names understood by the test-induction sidecar. The sample app sets
 * these on its outbound calls so the sidecar can select the right behavior.
 */
public final class InductionHeaders {

    /** Selects the active test profile (e.g. "payment-timeout"). */
    public static final String PROFILE = "x-induction-test-profile";

    /** Identifies the client type making the call (e.g. "payment-service"). */
    public static final String CALLER = "x-induction-test-caller";

    private InductionHeaders() {
    }
}

File: test-induction-api/src/main/scala/induction/InductionHeaders.scala
package induction

/** Headers a caller sends at request time to trigger a registered behavior.
  *
  *   - [[Profile]] selects which test profile is active (e.g. "payment-timeout").
  *   - [[Caller]]  identifies the client type making the call (e.g. "payment-service").
  *
  * The sidecar registers WireMock stubs that match on both of these, so the same
  * mock engine can serve different behaviors per profile and per client type.
  */
object InductionHeaders:
  val Profile = "x-induction-test-profile"
  val Caller  = "x-induction-test-caller"
